pkg/providers/aws/cluster_network_provider.go

```diff
@@ -26,77 +38,620 @@ import (
+	if !tagsContains(vpcTags, DefaultRHMIVpcNameTagKey, DefaultRHMIVpcNameTagValue) ||
+		!tagsContains(vpcTags, *clusterIDTag.Key, *clusterIDTag.Value) {
+
+		_, err := n.Ec2Api.CreateTags(&ec2.CreateTagsInput{
+			Resources: []*string{
+				aws.String(*vpc.VpcId),
+			},
+			Tags: []*ec2.Tag{
+				{
+					Key:   aws.String(DefaultRHMIVpcNameTagKey),
+					Value: aws.String(DefaultRHMIVpcNameTagValue),
+				}, clusterIDTag,
+			},
+		})
+		if err != nil {
+			return errorUtil.Wrap(err, "unable to tag vpc")
+		}
+		logger.Infof("successfully tagged vpc: %s for clusterID: %s", *vpc.VpcId, *clusterIDTag.Value)
+	}
+	return nil
+}
+
+//an rds subnet group is required to be in place when provisioning rds resources
+//
+//reconcileStandaloneRDSVpc ensures that an rds subnet group is created with 2 private subnets
+func (n *NetworkProvider) reconcileStandaloneRDSVpc(ctx context.Context, privateVPCSubnets []*ec2.Subnet, clusterID string) error {
+	logger := n.Logger.WithField("action", "reconcileStandaloneRDSVpc")
+	logger.Info("ensuring rds subnet groups in vpc are as expected")
+	// get subnet group id
+	subnetGroupName, err := BuildInfraName(ctx, n.Client, defaultSubnetPostfix, DefaultAwsIdentifierLength)
+	if err != nil {
+		return errorUtil.Wrap(err, "error building subnet group name")
+	}
+
+	foundSubnet, err := getRDSSubnetByGroup(n.RdsApi, subnetGroupName)
+	if foundSubnet != nil {
+		logger.Infof("subnet group %s found", *foundSubnet.DBSubnetGroupName)
+		return nil
+	}
+
+	// in the case of no private subnets being found, we return a less verbose error message compared to obscure aws error message
+	if len(privateVPCSubnets) == 0 {
+		return errorUtil.New("no private subnets found, can not create subnet group for rds")
+	}
+
+	// build array list of all vpc private subnets
+	var subnetIDs []*string
+	for _, subnet := range privateVPCSubnets {
+		subnetIDs = append(subnetIDs, subnet.SubnetId)
+	}
+
+	// build subnet group input
+	subnetGroupInput := &rds.CreateDBSubnetGroupInput{
+		DBSubnetGroupDescription: aws.String(defaultSubnetGroupDesc),
+		DBSubnetGroupName:        aws.String(subnetGroupName),
+		SubnetIds:                subnetIDs,
+		Tags: []*rds.Tag{
+			{
+				Key:   aws.String("cluster"),
+				Value: aws.String(clusterID),
+			},
+		},
+	}
+
+	// create db subnet group
+	logger.Infof("creating resource subnet group %s", *subnetGroupInput.DBSubnetGroupName)
+	if _, err := n.RdsApi.CreateDBSubnetGroup(subnetGroupInput); err != nil {
+		return errorUtil.Wrap(err, "unable to create db subnet group")
+	}
+	return nil
+}
+
+//It is required to have an elasticache subnet group in place when provisioning rds resources
+//
+//reconcileStandaloneElasticacheVpc ensures that an rds subnet group is created with 2 private subnets
+func (n *NetworkProvider) reconcileStandaloneElasticacheVpc(ctx context.Context, privateVPCSubnets []*ec2.Subnet) error {
+	logger := n.Logger.WithField("action", "configureElasticacheVpc")
```

Comment:
Looks like the action is incorrect here. I might look at a way to resolve this in my PR. Even using the action as more of a description may make sense.